docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,411 @@
+
+First, you'll need **Contentful**'s access tokens, of which you'll need three. These can all be found in the Settings menu, under the `API keys` option:
+
+- Space ID,
+- Delivery token,
+- Preview token;
+
+Each token must be used together with their specific host, as follows:
+
+- Delivery host: `cdn.contentful.com`,
+- Preview host: `preview.contentful.com`;
+
+Setting up the SDK can be done as follows: 
+
+```js
+import { createClient } from 'contentful';
+
+const client = createClient({
+    space: // Space ID token
+    accessToken: // Either the Delivery token, or the Preview token
+    host: // Either the Delivery host, or the Preview host
+});
+```
+
+This `client` exposes the functions necessary to fetch the data in **Contentful**, for example:
+
+```js
+const result = await client.getEntries({ 
+    content_type: 'clients',
+});
+```
+
+If you're taking advantage of localization in your **Contentful** space, you must add to the request a field with your intended language code, like so:
+
+```js
+const result = client.getEntries({
+    content_type: 'clients',
+    locale: 'en-US'
+});
+```
+
+### 2. Populating the app with fetched content
+
+To populate the app with the fetched content, we will explore two possible implementations: one for bigger applications that justify the inclusion of Redux store, and those which don't.
+
+#### 2.1. Accessing the client directly though `getInitialProps`
+
+In Next.js apps, the `getInitialProps` function in general components typically receives a context object, which exposes many aspects of the context in which the page is rendered. Because we are building a custom app, and doing ourselves the work of running these `getInitialProps` functions, we can control what data they receive. In practice, we will be populating the context object with the information we want to propagate to our components. So, in your `App.js` file, you can check whether the page was loaded with a `cms-preview` query parameter, build your client, and share it across the app.
+
+This function also has the advantage letting you return props to your App function, making it easier to signal the App itself if you're in preview mode.
+
+```js
+// App.js
+import { createClient } from 'contentful';
+
+App.getInitialProps = async ({Component, context, router}) => {
+
+    // Create default client
+    let client = createClient({
+        space: process.env.CONTENTFUL_SPACE_ID,
+        accessToken: process.env.CONTENTFUL_TOKEN, // Delivery API Token
+        host: 'cdn.contentful.com', // Delivery API host
+    });
+
+    // Check whether route has `cms-preview` query parameter
+    const isPreviewingContentful = Object.hasOwnProperty.call(router.query, 'cms-preview');
```

Comment:
Where does this `cms-preview` property come from exactly? By query parameter do you mean an actual URL query parameter like `?cms-preview=true`?

Reply:
Nevermind, I completely missed this in the documentation. 😄